### ISSUE TITLE
An addon that makes rounding corners in the costume editor easier

### ISSUE THREAD

**Issue by penguinmoose** [ScratchAddons/ScratchAddons#4999]
### Is your request related to a problem? Please describe.

Right now, the only way to make rounded corners in shapes in the costume editor is to use the point editor tool, make two rounded (Bézier curve) points an equal distance from the sharp corner per non-rounded point, then delete the sharp corners. This can be a bit hard and time-consuming - it isn’t easy to get all the rounded points an equal distance from the sharp corner.

I always need to make rounded corners in my projects, especially in buttons and pop-ups.

### Describe the solution you'd like

I think there should be an addon that does exactly that for me - making a shape in the costume editor have rounded corners. The addon can be a button next to all the options (forward, backward, bring to front, etc) on top of the canvas in the costume editor. Upon clicking the button, I’m thinking a slider would show in which you can choose how rounded corners should be.

### Describe alternatives you've considered

- A somewhat simpler solution would be to have a button next to “curved” and “pointed” shown when a pointed corner is selected that makes that specific point rounded (with an option controlling how much so).

### Additional context

None

**Comment by Samq64** [comment]
A workaround that I sometimes do is write the svg manually so I get 100% acceracy, but it's kind of a waste of time and tricky to do for anything more than a couple shapes.

**Comment by penguinmoose** [comment]
@Samq64 True. I only know a tiny bit about SVG code so I can’t do that. It would be nice to be able to round corners directly in the Scratch editor.

**Comment by Secret-chest** [comment]
The slider should be a pixel input

**Comment by mxmou** [comment]
> ### Describe alternatives you've considered
> * A somewhat simpler solution would be to have a button next to “curved” and “pointed” shown when a pointed corner is selected that makes that specific point rounded (with an option controlling how much so).

I think that's better because it works on individual corners, not entire shapes.

**Comment by Secret-chest** [comment]
> I guess I could just use inkscape, but meh.

Also remember that Inkscape imports are utterly broken most of the time if you do anything more than 1 shape

**Comment by Samq64** [comment]
> > I guess I could just use inkscape, but meh.
> 
> Also remember that Inkscape imports are utterly broken most of the time if you do anything more than 1 shape

Doesn't `fix-uploaded-svgs` fix that?

**Comment by Secret-chest** [comment]
> > > I guess I could just use inkscape, but meh.
> > 
> > Also remember that Inkscape imports are utterly broken most of the time if you do anything more than 1 shape
> 
> Doesn't `fix-uploaded-svgs` fix that?

no

**Comment by UtterDonkey** [comment]
> > ### Describe alternatives you've considered
> > 
> > * A somewhat simpler solution would be to have a button next to “curved” and “pointed” shown when a pointed corner is selected that makes that specific point rounded (with an option controlling how much so).
> 
> I think that's better because it works on individual corners, not entire shapes.

Yeah, there could be pointed, curved and rounded. When rounded is selected, an input box could appear where you can set the radius (like browser radius in CSS, but for just 1 point).
